File: Java/src/Patterns/Patterns.java
package Patterns;
import java.util.Scanner;
public class Patterns {
	public static void main(String[] args) {
		Scanner sc= new Scanner(System.in);
		System.out.print("enter a number:  ");
		int n=sc.nextInt();
//		pattern1(n);
//		pattern2(n);
//		pattern3(n);
//		pattern4(n);
//		pattern5(n);
//		pattern6(n);
//		pattern7(n);
//		pattern8(n); 
//		pattern9(n);
//		pattern10(n);
//		pattern11(n);
//		pattern12(n);
//		pattern13(n);
//		pattern14(n);
		pattern15(n);
//		pattern16(n);
		sc.close();

	}
	static void pattern1(int n) {				//	* * * * *
		for(int i=0;i<n;i++) {					//	* * * * *
			for(int j=0;j<n;j++) {				//	* * * * *
				System.out.print("* ");			//	* * * * *
			}									//	* * * * *
			System.out.println();
		}
	}
	static void pattern2(int n) {				//	*
		for(int i=1;i<=n;i++) {					//	* *
			for(int j=1;j<=i;j++) {				//	* * *
				System.out.print("* ");			//	* * * *
			}									//	* * * * *
			System.out.println();
		}
	}
	static void pattern3(int n) {				//	* * * * *
		for(int i=1;i<=n;i++) {					//	* * * *
			for(int j=n;j>=i;j--) {				//	* * *
				System.out.print("* ");			//	* *
			}									//	*
			System.out.println();
			
		}
	}
	static void pattern4(int n) {				//	*
		for(int i=1;i<=n;i++) {					//	* *
			for(int j=1;j<=i;j++) {				//	* * *
				System.out.print("* ");			// 	* * * *
			}									//	* * * * *
			System.out.println();				//	* * * *
		}										//	* * *
		for(int i=1;i<=n;i++) {					//	* *
			for(int j=n-1;j>=i;j--) {			//	*
				System.out.print("* ");
			}
			System.out.println();
		}
	}
	
	static void pattern5(int n) {				//	1
		for(int i=1;i<=n;i++) {					//	1 2
			for(int j=1;j<=i;j++) {				//	1 2 3
				System.out.print(j+" ");		//	1 2 3 4
			}									// 	1 2 3 4 5
			System.out.println();
		}
	}
	
	static void pattern6(int n) {				//	    *
		for(int i=1;i<=n;i++) {					//	   **
			for(int j=n-1;j>=i;j--) {			//	  ***
				System.out.print(" ");			//	 ****
			}									//	*****
			for(int j=1;j<=i;j++) {
				System.out.print("*");
			}
			System.out.println();
		}
	}
	
	static void pattern7(int n) {				//	*****
		for(int i=0;i<n;i++) {					//	 ****
			for(int j=0;j<i;j++) {				//	  ***
				System.out.print(" ");			//	   **
			}									//	    *
			for(int j=n;j>i;j--) {
				System.out.print("*");
			}
			System.out.println();
		}
	}
	
	
	static void pattern8(int n) {
		for(int i=1;i<=n;i++) {					//	    *
			for(int j=n-1;j>=i;j--) {			//	   * *
				System.out.print(" ");			//	  * * *
			}									//	 * * * *
			for(int j=1;j<=i;j++) {				//	* * * * *
				System.out.print("* ");
			}
			System.out.println();
		}
	}
	
	static void pattern9(int n) {				//	* * * * * 
		for(int i=0;i<n;i++) {					//	 * * * * 
			for(int j=0;j<i;j++) {				//	  * * * 
				System.out.print(" ");			//	   * *
			}									//	    *
			for(int j=n;j>i;j--) {
				System.out.print("* ");
			}
			System.out.println();
		}
	}
	
	static void pattern10(int n) {
		for(int i=0;i<n;i++) {
			for(int j=n-1;j>i;j--) {			//	    *
				System.out.print(" ");			//	   ***
			}									//	  *****
			for(int j=0;j<=i;j++) {				//	 *******
				System.out.print("*");			//	*********
			}
			for(int j=0;j<i;j++) {
				System.out.print("*");
			}
			System.out.println();
		}
	}
	
	static void pattern11(int n) {				//	*********
		for(int i=0;i<n;i++) {					//	 *******
			for(int j=0;j<i;j++) {				//	  *****
				System.out.print(" ");			//	   ***
			}									//	    *
			for(int j=n;j>i;j--) {	
				System.out.print("*");
			}
			for(int j=n-1;j>i;j--) {
				System.out.print("*");
			}
			System.out.println();
		}
	}
	
	
	static void pattern12(int n) {				//	* * * * *
		for(int i=0;i<n;i++) {					//	 * * * *
			for(int j=0;j<i;j++) {				//	  * * *
				System.out.print(" ");			//	   * * 
			}									//		*
			for(int j=n;j>i;j--) {				//	    *	
				System.out.print("* ");			//	   * *
			}									//	  * * *
			System.out.println();				//	 * * * *
		}										//	* * * * *
		for(int i=0;i<n;i++) {
			for(int j=n-1;j>i;j--) {
				System.out.print(" ");
			}
			for(int j=0;j<=i;j++) {
				System.out.print("* ");
			}
			System.out.println();
		}
	}
	
	static void pattern13(int n) {				//	* * * * *
		for(int i=0;i<n;i++) {					//	 * * * *
			for(int j=0;j<i;j++) {				//	  * * *
				System.out.print(" ");			//	   * * 
			}									//		*
			for(int j=n;j>i;j--) {				//	   * *
				System.out.print("* ");			//	  * * *
			}									//	 * * * *
			System.out.println();				//	* * * * *
		}										
		for(int i=1;i<n;i++) {
			for(int j=n-1;j>i;j--) {
				System.out.print(" ");
			}
			for(int j=0;j<=i;j++) {
				System.out.print("* ");
			}
			System.out.println();
		}
	}
	
	static void pattern14(int n) {				//	    *
		for(int i=1;i<=n;i++) {					//	   **
			for(int j=n;j>i;j--) {				//	  * *
				System.out.print(" ");			//	 *  *
			}									//	*****     
			for(int j=1;j<=i;j++) {
				if(j==1||j==i||i==n) {
					System.out.print("*");
				}
				else {
					System.out.print(" ");
				}
			}
			
			System.out.println();
			
		}
	}
	
	
	static void pattern15(int n) {				//	    *
		for(int i=1;i<=n;i++) {					//	   * *
			for(int j=1;j<=n-i;j++) {			//	  *   *
				System.out.print(" ");			//	 *     *
			}									//	*********
			if(i==1) {
				System.out.println("*");
			}
			else if(i<n) {
				System.out.print("*");
				for(int j=1;j<=2*i-3;j++) {
					System.out.print(" ");
				}
				System.out.println("*");
			}
			else {
				for(int j=1;j<=2*n-1;j++) {
					System.out.print("*");
				}
				System.out.println();
			}
			
			
		}
	}
	
	
}

File: Python/Patterns.py
def pattern1(n):  				 		#  * * * * *
    for i in range(n):           		#  * * * * *
        for j in range(n):       		#  * * * * *
            print("* ", end="")  		#  * * * * *
        print()                 		#  * * * * *

def pattern2(n):  						#  *
    for i in range(1, n+1):       		#  * *
        for j in range(1, i+1):   		#  * * *
            print("* ", end="")   		#  * * * *
        print()                   		#  * * * * *

def pattern3(n): 				  		#  * * * * *
    for i in range(1, n+1):      	 	#  * * * *
        for j in range(n, i-1, -1):   	#  * * *
            print("* ", end="")   		#  * *
        print()                   		#  *

def pattern4(n):  						#  *
    for i in range(1, n+1):         	#  * *
        for j in range(1, i+1):    	 	#  * * *
            print("* ", end="")     	#  * * * *
        print()                     	#  * * * * *
    for i in range(1, n+1):         	#  * * * *
        for j in range(n-1, i-1, -1):   #  * * *
            print("* ", end="")     	#  * *
        print()							#  *

def pattern5(n):  						#  1
    for i in range(1, n+1):         	#  1 2
        for j in range(1, i+1):     	#  1 2 3
            print(j, end=" ")      		#  1 2 3 4
        print()                     	#  1 2 3 4 5

def pattern6(n):  						#      *
    for i in range(1, n+1):         	#     **
        for j in range(n-1, i-1, -1):   #    ***
            print(" ", end="")      	#   ****
        for j in range(i):          	#  *****
            print("*", end="")
        print()

def pattern7(n):  						#  *****
    for i in range(n):             		#   ****
        for j in range(i):          	#    ***
            print(" ", end="")      	#     **
        for j in range(n, i, -1):   	#      *
            print("*", end="")
        print()

def pattern8(n):
    for i in range(1, n+1):         	#      *
        for j in range(n-1, i-1, -1):   #     * *
            print(" ", end="")      	#    * * *
        for j in range(i):          	#   * * * *
            print("* ", end="")     	#  * * * * *
        print()

def pattern9(n):  						#  * * * * *
    for i in range(n):             	 	#   * * * *
        for j in range(i):          	#    * * *
            print(" ", end="")      	#     * *
        for j in range(n, i, -1):   	#      *
            print("* ", end="")
        print()

def pattern10(n):
    for i in range(n):
        for j in range(n-1, i, -1):     #      *
            print(" ", end="")          #     ***
        for j in range(i+1):            #    *****
            print("*", end="")          #   *******
        for j in range(i):              #  *********
            print("*", end="")
        print()

def pattern11(n): 		 				#  *********
    for i in range(n):              	#   *******
        for j in range(i):          	#    *****
            print(" ", end="")      	#     ***
        for j in range(n, i, -1):   	#      *
            print("*", end="")
        for j in range(n-1, i, -1):
            print("*", end="")
        print()

def pattern12(n): 	 					#  * * * * *
    for i in range(n):              	#   * * * *
        for j in range(i):          	#    * * *
            print(" ", end="")      	#     * *
        for j in range(n, i, -1):   	#      *
            print("* ", end="")			#      *
        print()							#     * *
    for i in range(n):					#    * * *
        for j in range(n-1, i, -1):		#   * * * *
            print(" ", end="")			#  * * * * *
        for j in range(i+1):
            print("* ", end="")
        print()

def pattern13(n):  						#  * * * * *
    for i in range(n):              	#   * * * *
        for j in range(i):          	#    * * *
            print(" ", end="")      	#     * *
        for j in range(n, i, -1):   	#      *
            print("* ", end="")			#     * *
        print()							#    * * *
    for i in range(1, n):				#   * * * *
        for j in range(n-1, i, -1):		#  * * * * *
            print(" ", end="")
        for j in range(i+1):
            print("* ", end="")
        print()

def pattern14(n):  								#      *
    for i in range(1, n+1):         			#     **
        for j in range(n, i, -1):   			#    * *
            print(" ", end="")     	 			#   *  *
        for j in range(1, i+1):					#  *****
            if j == 1 or j == i or i == n:
                print("*", end="")
            else:
                print(" ", end="")
        print()

def pattern15(n):  						#      *
    for i in range(1, n+1):         	#     * *
        for j in range(n-i):        	#    *   *
            print(" ", end="")      	#   *     *
        if i == 1:						#  *********
            print("*")
        elif i < n:
            print("*", end="")
            for j in range(2*i-3):
                print(" ", end="")
            print("*")
        else:
            for j in range(2*n - 1):
                print("*", end="")
            print()

# Usage example
if __name__ == "__main__":
    # n = int(input("enter a number:  "))
    n=5
    # pattern1(n)
    # pattern2(n)
    # pattern3(n)
    # pattern4(n)
    # pattern5(n)
    # pattern6(n)
    # pattern7(n)
    # pattern8(n)
    # pattern9(n)
    # pattern10(n)
    # pattern11(n)
    # pattern12(n)
    # pattern13(n)
    # pattern14(n)
    pattern15(n)
